### PR TITLE
Add Wigner utility functions

### PR DESCRIPTION
**Pull request by gautierronan**
Adds the equivalent of `qutip.wigner` and `qutip.plot_wigner`.

### CODE REVIEW

File: torchqdynamics/utils/states.py [hunk 1]
```diff
@@ -92,7 +92,9 @@ def coherent(dim: int, alpha: complex | Tensor, *, dtype=None, device=None) -> T
                 [0.003+0.j],
                 [0.000+0.j]], dtype=torch.complex128)
     """
-    return displace(dim, alpha, dtype=dtype, device=device) @ fock(dim, 0)
+    return displace(dim, alpha, dtype=dtype, device=device) @ fock(
```
Comment (abocquet) [inline torchqdynamics/utils/states.py:95]: I'm guessing that implementing direclty the series expansion of a coherent state would be faster here. Is there a reason why we choose to apply the displacement operator to vacuum (other than being convenient)? 
If not maybe it's  an easy improvement that we could do sometimes

Comment (gautierronan) [reply torchqdynamics/utils/states.py:95]: From QuTiP's documentation:

> Select method 'operator' (default) or 'analytic'. With the 'operator' method, the coherent state is generated by displacing the vacuum state using the displacement operator defined in the truncated Hilbert space of size 'N'. This method guarantees that the resulting state is normalized. With 'analytic' method the coherent state is generated using the analytical formula for the coherent state coefficients in the Fock basis. This method does not guarantee that the state is normalized if truncated to a small number of Fock states, but would in that case give more accurate coefficients.

So there isn't a method that is always better than the other, but we could give the choice like qutip does.

**Review by abocquet — DISMISSED** [review]

File: torchqdynamics/utils/tensor_types.py [hunk 2]
```diff
@@ -99,11 +99,25 @@ def wrapper(
...
 ) -> torch.float32 | torch.float64:
     dtype = cdtype(dtype)
-    return torch.float64 if dtype is torch.complex128 else torch.float32
+    return DTYPE_CONVERSION_TABLE[dtype]
+
+
+def dtype_float_to_complex(
+    dtype: torch.float32 | torch.float64 | None = None,
+) -> torch.complex64 | torch.complex128:
+    return DTYPE_CONVERSION_TABLE[dtype]
```
Comment (pierreguilmin) [inline torchqdynamics/utils/tensor_types.py:120]: Fails if `None` is passed.

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
I don't have much time this week, so I'm not reviewing the content of `wigner.py`, I think it's quite independent of the rest of the library, so it's fine if we iterate on it later. One small important comment before merging (and removing `plot_wigner` as discussed).

File: torchqdynamics/utils/tensor_types.py (hunk 2 above)
Comment (gautierronan) [reply torchqdynamics/utils/tensor_types.py]: It's fixed. I also updated the name from `dtype_float_to_complex` to `dtype_real_to_complex` everywhere, to match the naming convention of pytorch.

File: torchqdynamics/utils/tensor_types.py [hunk 3]
```diff
@@ -99,11 +112,32 @@ def wrapper(
...
 
 
-def dtype_complex_to_float(
+DTYPE_TO_REAL = {
+    torch.complex64: torch.float32,
+    torch.complex128: torch.float64,
+    torch.float32: torch.float32,
+    torch.float64: torch.float64,
+}
+DTYPE_TO_CPLX = {
```
Comment (pierreguilmin) [inline torchqdynamics/utils/tensor_types.py:121]:
```suggestion
DTYPE_TO_COMPLEX = {
```

File: torchqdynamics/utils/tensor_types.py [hunk 4]
```diff
@@ -99,11 +112,32 @@ def wrapper(
     return wrapper
 
 
-def dtype_complex_to_float(
+DTYPE_TO_REAL = {
+    torch.complex64: torch.float32,
+    torch.complex128: torch.float64,
+    torch.float32: torch.float32,
+    torch.float64: torch.float64,
+}
```
Comment (pierreguilmin) [inline torchqdynamics/utils/tensor_types.py:120]:
```suggestion
DTYPE_TO_REAL = {
    torch.complex64: torch.float32,
    torch.complex128: torch.float64,
}
```

File: torchqdynamics/utils/tensor_types.py [hunk 5]
```diff
@@ -99,11 +112,32 @@ def wrapper(
...
+DTYPE_TO_CPLX = {
+    torch.float32: torch.complex64,
+    torch.float64: torch.complex128,
+    torch.complex64: torch.complex64,
+    torch.complex128: torch.complex128,
+}
```
Comment (pierreguilmin) [inline torchqdynamics/utils/tensor_types.py:126]:
```suggestion
    torch.float32: torch.complex64,
    torch.float64: torch.complex128,
}
```

**Pull request merged**